packages/config-loader/src/sources/ConfigSources.ts

```diff
@@ -144,6 +146,7 @@ export class ConfigSources {
     options: ConfigSourcesDefaultForTargetsOptions,
   ): ConfigSource {
     const rootDir = options.rootDir ?? findPaths(process.cwd()).targetRoot;
+    const watch = Boolean(options.watch ?? true);
```

Comment:
With the above change you should be able to skip this line and pass though `options.watch` as is